UsersApp/View Controllers/UserDetailsViewController.swift

```diff
@@ -0,0 +1,188 @@
+        NotificationCenter.default.removeObserver(UIResponder.keyboardWillChangeFrameNotification)
+    }
+    
+    private func setUpImageView(with urlString: String) {
+        userImageView.downloaded(from: urlString, contentMode: .scaleToFill)
+        userImageView.layer.cornerRadius = userImageView.frame.size.height / 2
+        userImageView.layer.masksToBounds = true
+    }
+    
+    private func setUpDetailView(_ view: UserDetailView ,_ title: String, _ subtitle: String!) {
+        view.contentViewTitle.text = title
+        view.contentViewSubtitle.text = subtitle
+    }
+    
+    private func setUpFirstNameView() {
+        setUpDetailView(firstNameView, "First Name", firstName)
+    }
+    
+    private func setUpLastNameView() {
+        setUpDetailView(lastNameView, "Last Name", lastName)
+    }
+    
+    private func setUpGenderView() {
+        setUpDetailView(genderView, "Gender", gender)
+    }
+    
+    private func setUpLocationView() {
+        setUpDetailView(cityView, "City", city)
+        setUpDetailView(stateView, "State", state)
+        setUpDetailView(streetAdressView, "Street Adress", streetAdress)
+        setUpDetailView(countryView, "Country", country)
+        setUpDetailView(postalCodeView, "Postal Code", postalCode?.description)
+        
+        let coordinatesSubtitle = "latitude: " + coordinatesLatitude! + "\nlongitude: " + coordinatesLongitude!
+        setUpDetailView(coordinatesView, "Coordinates", coordinatesSubtitle)
+        
+        let timezoneSubtitle = "offset: " + timezoneOffset! + "\n" + timezoneDescription!
+        setUpDetailView(timezoneView, "Timezone", timezoneSubtitle)
+    }
+    
+    private func setUpEmailView() {
+        setUpDetailView(emailView, "Email", email)
+    }
+    
+    private func setUpDobView() {
+        let dobSubtitle = "Date: " + dobDate! + ", age: " + String(dobAge!)
+        setUpDetailView(dateOfBirth, "Birth Date ", dobSubtitle)
+    }
+    private func setUpRegisteredView() {
+        let registeredSubtitle = "Date: " + registeredDate! + ", age: " + String(registeredAge!)
+        setUpDetailView(registeredDateView, "Registered ", registeredSubtitle)
+    }
+    
+    private func setUpPhoneView() {
+        setUpDetailView(phoneView, "Phone", phone)
+    }
+    
+    private func setUpCellphoneView() {
+        setUpDetailView(cellphoneView, "Cellphone", cellphone)
+    }
+    
+    //textView
+    
+    var userDefaultsKey: String {
+        guard let userId = dobDate else {
+            fatalError("User ID is nil")
+        }
+        return "text_\(userId)"
+    }
+    
+    let userDefaults = UserDefaults()
+    
+    func textViewDidChange(_ textView: UITextView) {
+        userDefaults.setValue(textView.text, forKey: userDefaultsKey)
+    }
+    
+    @objc private func handleTap(_ gesture: UITapGestureRecognizer) {
+        noteTextView.resignFirstResponder()
```

Comment:
for better user experience, the saving notes should be possible also from keyboard.
Also you can add "Cancel" and "Save" buttons